### ISSUE TITLE
if tumor sample is not purity, how to solve it?

### ISSUE THREAD

**Issue by moshl** [BoevaLab/FREEC#25]
Dear BoevaLab,
I have a mix tumor sample,including some normal cells, I know the percentage of tumor cells is 64%.
In config file, I set the parameters:" contamination=0.36 "and "contaminationAdjustment=TRUE", I don't whether it is ture. what is the algorithm for fixed the contamination? thanks. Looking forward to you reply.

**Comment by valeu** [comment]
Dear user,
If you know that it is 64%, then you are write to set "contamination=0.36 "and "contaminationAdjustment=TRUE". If you have doubts, you can set "contaminationAdjustment=TRUE". But in this case you can risk that FREEC will make an error in the evaluation of contamination. In our tests, when the contamination is higher than 50% FREEC is likely to make errors.

**Comment by moshl** [comment]
Thanks valeu very much. I get it . So, what is the algorithm for adjustment according to the contamination? looking forward to your reply.

**Comment by valeu** [comment]
yes. If you do not provide "contamination", it will evaluate it itself.

**Comment by moshl** [comment]
i am sorry. my questions is ：what is the math model used for adjustment？if tumor cells rate is very low, what should i do?

**Comment by valeu** [comment]
FREEC uses regression on the normalized ratio values in regions of predicted gain and loss. However, if the contamination is higher than 50% many (or even most) of such gain/loss regions will be missed by FREEC and the evaluation of the contamination will fail. The only solution I see for such samples is to set "contamination" yourself and check whether the resulting profile looks OK. i.e., whether copy number aberrations fall on integer values (1, 2, 3, etc.)

**Comment by moshl** [comment]
Thanks very much. You said "In our tests, when the contamination is higher than 50% FREEC is likely to make errors."  Did those results in tests  publish? In my samples, the truely rate of tumor cells is 17%, I set the contamination=83. The CNV result in this sample is very strange.  the attach is my results. So, What should I set the different contamination?





![lxf_1d8 mpileup_baf txt](https://user-images.githubusercontent.com/29252072/27685489-f5d7f852-5d00-11e7-8fe9-abb713bb7da3.png)
![lxf_1d8 mpileup_ratio txt](https://user-images.githubusercontent.com/29252072/27685527-1b5a2938-5d01-11e7-971a-3f438c26cb99.png)
![Uploading Lxf_1D8.mpileup_ratio.txt.png…]()
